modules/lang-painless/src/main/java/org/elasticsearch/painless/Definition.java

```diff
@@ -577,7 +583,13 @@ private Definition(List<Whitelist> whitelists) {
 
         // goes through each Painless struct and determines the inheritance list,
         // and then adds all inherited types to the Painless struct's whitelist
-        for (Struct painlessStruct : structsMap.values()) {
+        for (String painlessStructName : structsMap.keySet()) {
```

Comment:
What's the reason for moving to key iteration followed by lookup and filtering? Iterating values + filtering should be faster (not that performance matters here really, but...).

Reply:
Was for more explicit naming, but changed to use entryset to get the values from.